aci/layout.go

```diff
@@ -118,9 +118,16 @@ Tar:
 			}
 			rfsOK = true
 		default:
-			flist = append(flist, name)
+			if _, seen := fseen[name]; seen {
+				return fmt.Errorf("Duplicate file entry in archive: %s", name)
+			}
+			fseen[name] = true
 		}
 	}
+	var flist []string
```

Comment:
random sidenote: it bothers me that there isn't a better way to preallocate here, unless I'm just too dumb to think of it